### DISCUSSION TITLE
Clarification on data preparation and detection model training

### DISCUSSION THREAD

**Discussion by ep0p** [mindee/doctr#1653] [Q&A] [answered]
I am currently collecting and preparing my data in order to train a detection model and something is unclear for me. In the json i need to pass the bbox coordinates of the words in the image organized in lines and then blocks. 

Could you please clarify what the [classes](https://github.com/mindee/doctr/tree/main/references/detection#readme) are in this context?

**Comment by felixdittrich92** [comment]
Hi @ep0p :wave:;

The last example snippet is if you want to train a KIE model.
So for pure text detection you need the above sinppet:
```
{
    "sample_img_01.png" = {
        'img_dimensions': (900, 600),
        'img_hash': "theimagedumpmyhash",
        'polygons': [[[x1, y1], [x2, y2], [x3, y3], [x4, y4]], ...]
     },
     "sample_img_02.png" = {
        'img_dimensions': (900, 600),
        'img_hash': "thisisahash",
        'polygons': [[[x1, y1], [x2, y2], [x3, y3], [x4, y4]], ...]
     }
     ...
}
```
Attachment as reference / example file:  (NOTE: labels are not required this was only added to create easier a recognition dataset from it :) )

https://drive.google.com/file/d/1CHVd2B4J4ejbwum0SCV5xD8Flj2UOqNX/view?usp=drive_link

Best regards,
Felix

**Reply by felixdittrich92** [reply, marked as answer]
NOTE: 
polygons are the 4 corner points for each `word` so there are no annotations for `lines` or `blocks` necessary / required.
- absolute values / Integers

You need one folder `train` and one folder `val` with a folder `images` and `labels.json` in each.

**Reply by felixdittrich92** [reply]
https://github.com/mindee/doctr/discussions/1654

**Reply by ep0p** [reply]
Hi @felixdittrich92,

Thank you for clarification. 

That was stupid from my part: i misunderstood the nested list as lines and then blocks. Now, while i can have the bbox coords for lines i do not have them for words. I am starting to reconsider the need for training for detection. 

My dataset mostly contains: the docs, pdfs (which i will have to split into images), and the text. So i guess that the training should be done mostly on recognition in my case.

Does this seem like the right approach, or do you have any other suggestions? I am open to any advice.

**Reply by felixdittrich92** [reply]
If our default provided detection model works well in your case and you want only to improve the recognition (string for each detection) or train on another vocabulary (language) then yes recognition training would be the correct one :)

NOTE: For recognition training you need also the word coordinates or already cropped word images (take a look into the provided reference datasets i have shared in #1654)

**Reply by ep0p** [reply]
Alright, thank you for the clarification and the datasets
